httpclient/httpclient.go

```diff
@@ -94,6 +104,12 @@ func (c *Config) Canonicalize() {
 			c.AuthURL = "https://auth.hashicorp.com"
 		}
 	}
+	// Allow https:// prefix on HostPath even though it's the only scheme we allow
+	// as it's more natural to support the URL. Any other scheme we don't strip
+	// which will fail validation.
+	if strings.HasPrefix(strings.ToLower(c.HostPath), "https://") {
+		c.HostPath = c.HostPath[8:]
+	}
```

Comment:
If anyone disagrees with this change I don't have a strong opinion and happy to revert but it seems convenient to allow this - it could be more intuitive for users to not need to remember to specify a protocol in one ENV var but not in the other for example...

Reply:
One other question that puzzled me: is there a reason that we set defaults for these two configs here in `Canonicalize` but then we set the oauth secrets from ENV in `Validate`? It seems like those should maybe be in here too if they mutate the config?

I didn't do that in this PR but happy to if that would be helpful!

Reply:
I'll leave it in this PR to keep things constrained. Would you like me to create another with that change?

Reply:
Sure!